tsdb/engine/tsm1/tombstone.go

```diff
@@ -50,6 +52,9 @@ type Tombstoner struct {
 	pendingFile       *os.File
 	tmp               [8]byte
 	lastAppliedOffset int64
+
+	id  uint64
```

Comment:
I notice that you added the shard id in FileStore as well. I suggest either add shard ID to the Tombstoner and FileStore constructor so that it's always available, or add it to the observer implementation, with a new instance per shard. I favor the latter.